### ISSUE TITLE
dynamically adding $facets

### ISSUE THREAD

**Issue by marbemac** [Yomguithereal/baobab#313]
I remember a while back this was mentioned somewhere, but I can't seem to find it. Is it possible to add a facet to the tree after instantiation?

```js
var tree = new Baobab({
  projects: {
    123: {name: "foo"}
  }
}, {})

tree.set('currentProject', {
  id: 123,
  $data: {
    cursors: {
      id: ['currentProject', 'id'],
      projects: ['projects']
    },
    get: function(data) {
      return data.projects[data.id] || {}
    }
  }
})

// This returns the object defined above (cursors and get function), not the 
// computed / resolved data from the get function.
tree.get(['currentProject', '$data'])
```

Thanks!

**Comment by Yomguithereal** [comment]
This is something I must add but as it is not trivial to setup at all I went to fix other things. But this is definitely on the roadmap.

**Comment by marbemac** [comment]
Ah ok, thanks for the response. I can work around this for now.

**Comment by hyusetiawan** [comment]
@Yomguithereal any update on this? 
@marbemac how do you get around it? Without defining facets upon instantiation, it seems this is the only way but we would like to have each of our components to define their own facets from the state

**Comment by Yomguithereal** [comment]
@hyusetiawan, the only reason why you would use a facet normally is because multiple components want to listen to the same view over your static data (for convenience and performance reasons). If each of your components needs to format the data in its own way, they I would say you would rather compute the data into the component itself, somewhat.

**Comment by hyusetiawan** [comment]
The way i am using it right now is by having a set of basic data that each component derive meaning from it through the use of facets. This has been working really well but severely stunted by the fact that all of those facets need to be instantiated at once, delegating that to components would be much cleaner in my opinion, is it not part of the planned feature? (I read that it is, somewhere)

**Comment by Yomguithereal** [comment]
Setting/updating/deleting facets at runtime is indeed a planned feature and will normally ship with a stable v2 version of the lib. However, for what you seem to want, i.e. making components treat the bound data to fit their needs, why can't you do it within the render function, for instance?

**Comment by marbemac** [comment]
What I was considering doing is using dynamic facets to essentially create relationships between models. 

I have a state tree with a map of users and a map of organizations. A user belongs to an organization. When a new user is added to the tree I'd like to set their $organization prop to a facet that pulls in the org info from the organizations map.

For now, I've reversed the process. I've created a listener to the organizations map, that updates any corresponding users when an organization changes.
